cmake/std/trilinosprhelpers/setenvironment/SetEnvironment.py

```diff
@@ -0,0 +1,447 @@
+        print("+--------------+")
+
+        for envvar in self.actions["setenv"]:
+            k = envvar['key']
+            v = envvar['value']
+            print("[envvar] export {}={}".format(k,v))
+            v = self._expand_envvars_in_string(v)
+            os.environ[k] = v
+            print("[envvar] export {}={} (actual)".format(k,v))
+        for k in self.actions["unsetenv"]:
+            del os.environ[k]
+            print("[envvar] unset {}".format(k))
+
+        if status != 0 and throw_on_error:
+            sys.stdout.flush()
+            raise Exception("FATAL ERROR in SetEnvironment.apply()")
+
+        return status
+
+
+    def pretty_print_envvars(self, envvar_filter=None, filtered_keys_only=False):
+        """
+        Print out a filtered list of environment variables. This routine provides
+        a simplified view of the environment variables on a system since sometimes
+        just printing out the contents of all envvars on a system can result in very
+        cluttered console logs that are difficult to read through.
+
+        `envvar_filter` contains a list of strings that is matched against envvar
+        key names, if the envvar key contains one of the strings in envvar_filter
+        then we print both the envvar and its value. Otherwise, we print just the
+        envvar name.
+
+        If we set filtered_keys_only to True then we ONLY print out envvars that
+        matched envvar_filter.
+
+        Arguments:
+            envvar_filter (list): a list of keys to print out the value.
+                all envvar values are printed if omitted.
+                Default: None
+            filtered_keys_only (bool): If true, we only display filtered keys.
+                If false, we display the keys of all keys.
+                Default: False
+
+        Returns:
+            int 0
+
+        Todo:
+            This could probably be moved outside of this class since it's not
+            really directly relevant to the environment setting stuff, but it
+            might also be useful to have here for checking our work.
+        """
+        if envvar_filter is not None:
+            assert isinstance(envvar_filter, list)
+
+        print("+" + "="*38 + "+")
+        print("|   P R I N T   E N V I R O N M E N T   V A R S")
+        print("+" + "="*38 + "+")
+        print("--- ")
+        for k,v in os.environ.items():
+            matched_key = False
+            if envvar_filter is not None:
+                for f in envvar_filter:
+                    if f in k:
+                        matched_key = True
+                        break
+            else:
+                filtered_keys_only = False
+
+            if filtered_keys_only == False or matched_key:
+                print("--- {}".format(k), end="")
+                if envvar_filter is None:
+                    print(" = {}".format(v), end="")
+                elif matched_key:
+                    print(" = {}".format(v), end="")
```

Comment:
I don't understand this if-block.  Looks like it's doing the same thing in either case.

Reply:
the default argument to envvar_filter is None.  so you can call `pretty_print_envvars()` with no parameters and it should work.

This if/elif block could have also been written:

```python
if envvar_filter is None or matched_key:
    print(...)
```

but I broke it out into an if / elif to make it easier to do something different if I want to change it later.  This is a habit actually from my work as a compiler developer back in the day.  I tend to unroll things a bit more since compilers will do the fusion for you, but it is often easier to understand code that doesn't overly complicate the expression or if people try and put too much into a loop.  

In terms of performance, there's little difference between fusing this into a single if statement vs. having the if..elif.

The general operation to this function is something like:
```
for each item in the environment vars:
    for each item in the list of filters:
        if the envvar name contains the filter item:
            set matched_key = true and break
    print the envvar if it should be printed and its value if that should be printed
```